From c487c1e3fbfe58860f19902628e9f13f69b7cff3 Mon Sep 17 00:00:00 2001
From: jianlonghuang <jianlong.huang@starfivetech.com>
Date: Wed, 9 Mar 2022 15:00:25 +0800
Subject: [PATCH 34/34] Resolved issue #703 add tag info

Signed-off-by: jianlonghuang <jianlong.huang@starfivetech.com>
---
 scripts/setlocalversion | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/scripts/setlocalversion b/scripts/setlocalversion
index 4a63143706..3cb766ef26 100755
--- a/scripts/setlocalversion
+++ b/scripts/setlocalversion
@@ -66,6 +66,10 @@ scm_version()
 			else
 				printf '%s%s' -g $head
 			fi
+		else
+			atag="`git describe 2>/dev/null`"
+			printf '%s%s' - ${atag}
+			#printf '%s%s' -g $head
 		fi
 
 		# Is this git on svn?
-- 
2.37.0

